src/hdx_cli_toolkit/utilities.py

```diff
@@ -335,3 +335,31 @@ def print_dictionary_comparison(
                 flush=True,
             )
     print("-" * total_width, flush=True)
+
+
+def query_dict(
+    key_: str,
+    output: list[dict[str, Any]],

```

Comment:
Instead I made the query_dict function return the whole list of output rows for a dataset. Invoking the function one key at a time won't work because each key gets a new row.